src/sdk/namespace_fs.js

```diff
@@ -665,14 +670,20 @@ class NamespaceFS {
                 }
                 const marker_dir = key_marker.slice(0, dir_key.length);
                 const marker_ent = key_marker.slice(dir_key.length);
+                // dir_key and marker_dir comparison will fail when there are folder with structure slimier to 
+                // "dir_prfix.12345/" and "dir_prfix.12345.old/" because "/" considered greater than "."  to fix this 
+                // all the backslash is replaced with space. This updated marker and dir_key used only for comparison.
+                const updated_marker_dir = marker_dir.replaceAll('/', ' ');
```

Comment:
I think that instead of replacing the value of / we need to have a better comparison function, instead of comparing dir_key < marker_dir and so on, we will call that comparison function. WDYT?